File: 1768-merge-strings-alternately/1768-merge-strings-alternately.cs
public class Solution {
    public string MergeAlternately(string word1, string word2) {
        // start from the first Word
        bool FirstTurn = true;
        
        int ptr1 = 0;
        int ptr2 = 0;
        char[] chars = new char[word1.Length + word2.Length];

        while (ptr1 < word1.Length && ptr2 < word2.Length){
            if (FirstTurn){
                chars[ptr1 + ptr2] = word1[ptr1];
                ptr1++;
            }
            else
            {
                chars[ptr1 + ptr2] = word2[ptr2];
                ptr2++;
            }
            FirstTurn = !FirstTurn;
        }

        while (ptr1 < word1.Length){
            chars[ptr1 + ptr2] = word1[ptr1];
            ptr1++;
        }

        while (ptr2 < word2.Length){
            chars[ptr1 + ptr2] = word2[ptr2];
            ptr2++;
        }

        return new string(chars);
    }
}

File: 1768-merge-strings-alternately/1768-merge-strings-alternately.py
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:        
        ptr1 = 0
        ptr2 = 0
        word1_len = len(word1)
        word2_len = len(word2)
        word1_turn = True
        ptr = 0
        res = [None for _ in range(word1_len + word2_len)]
        
        while ptr1 < word1_len and ptr2 < word2_len:
            if word1_turn:
                res[ptr] = word1[ptr1]
                ptr1 += 1
            else:
                res[ptr] = word2[ptr2]
                ptr2 += 1
                
            ptr += 1
            word1_turn = not word1_turn

        while ptr1 < word1_len:
            res[ptr] = word1[ptr1]
            ptr1 += 1
            ptr += 1

        while ptr2 < word2_len:
            res[ptr] = word2[ptr2]
            ptr2 += 1
            ptr += 1

        return "".join(res)